pallets/pallet-dyn-filter/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+		type EnsureOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
+
+		/// Governance calls. (GovernanceCall contains all system calls, return true if call is contained in this set)
+		type GovernanceCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// Staking, delegating, collating calls. (return true if call is contained in this set)
+		type StakingCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// System calls are not filtered. (return true if call is contained in this set)
+		type TransferCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// System calls are not filtered. (return true if call is contained in this set)
+		type FeatureCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// System calls are not filtered. (return true if call is contained in this set)
+		type XcmCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// System calls are not filtered. (SystemCall contains all system calls, return true if system call)
+		type SystemCall: Contains<<Self as frame_system::Config>::Call>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	#[pallet::getter(fn connected_accounts)]
+	pub type Filter<T> = StorageValue<_, FilterSettings, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		NewFilterRules { rules: FilterSettings },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(<T as Config>::WeightInfo::set_filter())]
+		pub fn set_filter(origin: OriginFor<T>, filter: FilterSettings) -> DispatchResult {
+			<T as Config>::EnsureOrigin::ensure_origin(origin)?;
+
+			Filter::<T>::set(filter);
+			Self::deposit_event(Event::<T>::NewFilterRules { rules: filter });
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Contains<T::Call> for Pallet<T> {
+		fn contains(t: &T::Call) -> bool {
+			// true = call goes through
+
+			// System relevant calls cannot be filtered
+			if T::SystemCall::contains(t) {
+				return true;
+			}
+
+			let FilterSettings {
+				governance,
+				staking,
+				transfer,
+				feature,
+				xcm,
+			} = Filter::<T>::get();
+			if governance && T::GovernanceCall::contains(t) {
+				return false;
+			}
+			if staking && T::StakingCall::contains(t) {
+				return false;
+			}
+			if transfer && T::TransferCall::contains(t) {
+				return false;
+			}
+			if feature && T::FeatureCall::contains(t) {
+				return false;
+			}
+			if xcm && T::XcmCall::contains(t) {
+				return false;
+			}
+			true
+		}
+	}
+
+	impl<T: Config> EnabledFunctionality for Pallet<T> {
+		fn governance() -> bool {
+			let FilterSettings { governance, .. } = Filter::<T>::get();
+			return governance;
```

Comment:
Why is the `return` statement required here?

Reply:
Sloppy me

Reply:
[7d2ddc0](https://github.com/KILTprotocol/kilt-node/pull/397/commits/7d2ddc05ea3d21a15a0f766444934b2c8c37ac97)